### PR TITLE
fix: dispatch safearea inset setter logic

### PR DESCRIPTION
**Pull request by kazo0**
relates to https://github.com/unoplatform/uno/issues/12007

depends on https://github.com/unoplatform/uno/pull/12246

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the current behavior?

Screenshot_1683741881

## What is the new behavior?

Screenshot_1683741883

### CODE REVIEW

File: src/Uno.Toolkit.UI/Helpers/DispatcherCompat.cs
Comment (Xiaoy312) [inline src/Uno.Toolkit.UI/Helpers/DispatcherCompat.cs:1]: so in the end, we dont need to override `HasThreadAccess`?

File: src/Uno.Toolkit.RuntimeTests/Tests/SafeAreaTests.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+{
+	[TestClass]
+	[RunsOnUIThread]
+	internal partial class SafeAreaTests
+	{
+#if __ANDROID__
+		[TestMethod]
+		public async Task Translucent_SystemBars()
+		{
+			using var _ = UseFullWindow();
```
[the remark this answers is not in the document]
Comment (kazo0) [reply src/Uno.Toolkit.RuntimeTests/Tests/SafeAreaTests.cs:49]: No, it sets the [full screen window attributed on android](https://github.com/unoplatform/uno.ui.runtimetests.engine/blob/cbf16c3685b2dc56161f495586b29e8f17d6cbf6/src/Uno.UI.RuntimeTests.Engine.Library/UI/UnitTestsControl.cs#LL783C1-L786C7) which it probably shouldn't do and that causes the status bar and nav bars to be hidden

File: src/Uno.Toolkit.UI/Controls/SafeArea/SafeArea.cs [hunk 2]
```diff
@@ -562,24 +560,35 @@ private void ApplyInsets(Thickness insets)
...
-
-				if (_insetMode == InsetMode.Padding
-					&& !PaddingHelper.GetPadding(owner).Equals(insets)
-					&& PaddingHelper.SetPadding(owner, insets))
-				{
-					_appliedPadding = insets;
-					LogApplyInsets();
-				}
-				else if (_insetMode == InsetMode.Margin)
+#if __ANDROID__
```
[the remark this answers is not in the document]
Comment (kazo0) [reply src/Uno.Toolkit.UI/Controls/SafeArea/SafeArea.cs:563]: I believe yes I can just dispatch the `InvalidateMeasure`, I will change it

File: src/Uno.Toolkit.UI/Helpers/DispatcherCompat.cs
Comment (kazo0) [reply src/Uno.Toolkit.UI/Helpers/DispatcherCompat.cs:1]: Well I created a new `Schedule` method that doesn't check `HasThreadAccess` and I use that in the SafeArea logic

File: src/Uno.Toolkit.RuntimeTests/Tests/SafeAreaTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (Xiaoy312) [reply src/Uno.Toolkit.RuntimeTests/Tests/SafeAreaTests.cs:49]: can you add this answer in comment on `UseFullWindow`?

File: src/Uno.Toolkit.UI/Controls/SafeArea/SafeArea.cs [hunk 3]
```diff
@@ -562,10 +560,9 @@ private void ApplyInsets(Thickness insets)
 				{
 					return;
 				}
-
-				if (_insetMode == InsetMode.Padding
-					&& !PaddingHelper.GetPadding(owner).Equals(insets)
-					&& PaddingHelper.SetPadding(owner, insets))
+				if (_insetMode == InsetMode.Padding &&
+					!PaddingHelper.GetPadding(owner).Equals(insets) &&
+					PaddingHelper.SetPadding(owner, insets))
```
Comment (Xiaoy312) [inline src/Uno.Toolkit.UI/Controls/SafeArea/SafeArea.cs:565]: you can create a new method `TryUpdatePadding` (TrySet was taken) that checks for inequality and then assigns value
currently we are doing reflection(even if memorized) on the property twice shortly when that could be avoided

**Review by Xiaoy312 — APPROVED** [review]

File: src/Uno.Toolkit.UI/Controls/SafeArea/SafeArea.cs [hunk 4]
```diff
@@ -580,6 +577,11 @@ private void ApplyInsets(Thickness insets)
 					}
 				}
 
+#if __ANDROID__
+				// Dispatching on Android prevents issues where layout/render changes occurring
+				// during initial loading of the view are not always properly picked up by the layouting/rendering engine.
```
Comment (agneszitte) [inline src/Uno.Toolkit.UI/Controls/SafeArea/SafeArea.cs:582]:
```suggestion
				// Dispatching on Android prevents issues where layout/render changes, occurring
				// during the initial loading of the view, are not always properly picked up by the layouting/rendering engine.
```

File: src/Uno.Toolkit.UI/Controls/SafeArea/SafeArea.cs [hunk 5]
```diff
@@ -652,21 +654,24 @@ internal void OnInsetModeChanged(InsetMode oldValue, InsetMode newValue)
...
+							owner.Margin = _originalMargin;
+						}
+						else if (oldValue == InsetMode.Padding)
+						{
+							_appliedPadding = new Thickness(0);
+							PaddingHelper.SetPadding(owner, _originalPadding);
+						}
+#if __ANDROID__
+						// Dispatching on Android prevents issues where layout/render changes occurring
+						// during initial loading of the view are not always properly picked up by the layouting/rendering engine.
```
Comment (agneszitte) [inline src/Uno.Toolkit.UI/Controls/SafeArea/SafeArea.cs:669]:
```suggestion
						// Dispatching on Android prevents issues where layout/render changes, occurring
						// during the initial loading of the view, are not always properly picked up by the layouting/rendering engine.
```

**Comment by Xiaoy312** [comment]
@kazo0 we should be able to move forward with this (with package bump)? as its dependency got merged

**Review by spouliot — APPROVED** [review]

**Pull request merged**